tests/test_user_prompt.py

```diff
@@ -0,0 +1,327 @@
+    """check that prompt_user_for_input correctly returns the users input and that other options work as expected"""
+
+    # Arrange
+    def mock_user_input_error_side_effect(*args, **kwargs):
+        raise errors.UserInput("test error")
+
+    side_effect_by_user_input = {
+        'dir': ['dir', 'Q'],
+        'error': mock_user_input_error_side_effect
+    }
+    patch_for_input = mocker.patch("builtins.input",
+                                   side_effect=side_effect_by_user_input.get(user_input, [user_input]))
+
+    patch_for_exit = mocker.patch("sys.exit")
+    current_dir_path = Path.cwd()
+    patch_for_get_base_dir = mocker.patch("RUFAS.user_prompt.Utility.get_base_dir", return_value=current_dir_path)
+
+    # Act
+    actual_user_input = prompt_user_for_input()
+
+    # Assert
+    captured = capsys.readouterr().out.split('\n')
+
+    if user_input in ["Q", "q"]:
+        patch_for_input.assert_called_once_with("\nEnter RUFAS Input: ")
+        patch_for_exit.assert_called_once()
+        assert "Exiting RUFAS..." in captured
+    elif user_input == "dir":
+        assert patch_for_input.call_count == 2
+        patch_for_get_base_dir.assert_called_once()
+        patch_for_exit.assert_called_once()
+        assert "Exiting RUFAS..." in captured
+        assert str(current_dir_path) in captured
+    elif user_input == "error":
+        assert 'USER INPUT ERROR: test error' in captured
+    else:
+        patch_for_input.assert_called_once_with("\nEnter RUFAS Input: ")
+        assert actual_user_input == 'input/' + user_input
+
+
+@pytest.mark.parametrize("input_path, make_graphs, verbose", [
+    ("dummy_path", True, True),
+    ("dummy_path", False, True),
+    ("dummy_path", True, False),
+    ("dummy_path", False, False)
+])
+def test_run_rufas(input_path: str,
+                   make_graphs: bool,
+                   verbose: bool,
+                   mocker: MockerFixture) -> None:
+    """Checks that run_rufas() calls the correct functions in the correct order"""
+    # Arrange
+    patch_for_set_global_variables = mocker.patch("main.set_global_variables")
+    file_list = ['file1.json', 'file2.json']
+    patch_for_obtain_file_list = mocker.patch("main.obtain_file_list", return_value=file_list)
+    patch_for_execute_simulations_from_files = mocker.patch("main.execute_simulations_from_files")
+
+    # Act
+    run_rufas(input_path, make_graphs, verbose)
+
+    # Assert
+    patch_for_set_global_variables.assert_called_once_with(make_graphs, verbose)
+    patch_for_obtain_file_list.assert_called_once_with(input_path, verbose)
+    patch_for_execute_simulations_from_files.assert_called_once_with(file_list)
+
+
+@pytest.mark.parametrize("make_graphs, verbose", [
+    (True, True),
+    (False, True),
+    (True, False),
+    (False, False)
+])
+def test_set_global_variables(make_graphs: bool,
+                              verbose: bool,
+                              mocker: MockerFixture) -> None:
```

Comment:
@ltp23, it also seems that mocker is not used here. Can I remove it safely?
```suggestion
def test_set_global_variables(make_graphs: bool,
                              verbose: bool) -> None:
```

Reply:
Thank you. I removed the unused mocker parameter.